test/functional/bip65-cltv-p2p.py

```diff
@@ -33,19 +26,34 @@ def cltv_invalidate(tx):
     tx.vin[0].scriptSig = CScript([OP_1NEGATE, OP_CHECKLOCKTIMEVERIFY, OP_DROP] +
                                   list(CScript(tx.vin[0].scriptSig)))
 
+def cltv_validate(node, tx, height):
+    '''Modify the signature in vin 0 of the tx to pass CLTV
+    Prepends <height> CLTV DROP in the scriptSig, and sets
+    the locktime to height'''
+    tx.vin[0].nSequence = 0
+    tx.nLockTime = height
+
+    # Need to re-sign, since nSequence and nLockTime changed
+    signed_result = node.signrawtransaction(ToHex(tx))
+    new_tx = CTransaction()
+    f = BytesIO(hex_str_to_bytes(signed_result['hex']))
+    new_tx.deserialize(f)
 
-class BIP65Test(ComparisonTestFramework):
+    new_tx.vin[0].scriptSig = CScript([CScriptNum(height), OP_CHECKLOCKTIMEVERIFY, OP_DROP] +
+                                  list(CScript(new_tx.vin[0].scriptSig)))
+    return new_tx
 
+class BaseNode(NodeConnCB):
     def __init__(self):
         super().__init__()
-        self.num_nodes = 1
-        self.extra_args = [['-whitelist=127.0.0.1', '-blockversion=3']]
 
-    def run_test(self):
-        test = TestManager(self, self.options.tmpdir)
-        test.add_all_connections(self.nodes)
-        NetworkThread().start() # Start up network handling in another thread
-        test.run()
+class BIP65Test(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 1
+        self.extra_args = [['-promiscuousmempoolflags=1', '-whitelist=127.0.0.1']]
+        self.setup_clean_chain = True
 
     def create_transaction(self, node, coinbase, to_address, amount):
```

Comment:
This doesn't need to be a method in `BIP65Test`. Consider moving it above the class definition to be a module-level function (alongside `cltv_invalidate()` and `cltv_validate()`)